### PR TITLE
Self review form ui

### PR DESCRIPTION
**Pull request by RakshandaWankhede**
Self review add update delete details create done

### CODE REVIEW

File: client/src/components/Employee/EmployeeListing.js [hunk 1]
```diff
@@ -169,20 +169,57 @@ const EmployeeListing = props => {
...
-              }}
-              inputProps={{
-                onChange: changeHandler,
-                placeholder: 'Search Employee',
-                inputProps: {
-                  'aria-label': 'Search'
-                }
-              }}
-            />
+          <>
```
Comment (shivamsharma30) [inline client/src/components/Employee/EmployeeListing.js:172]: Please break this code into sub components.

File: client/src/components/Employee/EmployeeListing.js [hunk 2]
```diff
@@ -169,20 +169,57 @@ const EmployeeListing = props => {
...
+            <GridItem xs={12} sm={12} md={12}>
+              <CustomInput
+                formControlProps={{
+                  className: classes.margin + ' ' + classes.search
```
Comment (shivamsharma30) [inline client/src/components/Employee/EmployeeListing.js:176]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/components/Employee/EmployeeListing.js [hunk 3]
```diff
@@ -192,43 +229,41 @@ const EmployeeListing = props => {
...
+                          : null
+                      }
+                      tableData={employeeDetails || null}
+                      addLinks={links}
+                      updateUser={updateUser}
+                      deleteUser={deleteUser}
+                      showLink={false}
+                    />
+                  ) : (
+                      '**Please search for employee result'
```
Comment (shivamsharma30) [inline client/src/components/Employee/EmployeeListing.js:253]: Please use Paragraph element.

File: client/src/components/Employee/EmployeeListing.js [hunk 4]
```diff
@@ -169,20 +169,57 @@ const EmployeeListing = props => {
...
```
Comment (shivamsharma30) [inline client/src/components/Employee/EmployeeListing.js:218]: Please use Paragraph element.

File: client/src/components/SelfReviewForm/SelfReviewForm.js [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+    }
+  }, [selfReviewUpdateError, addToast, dispatch])
+  const submitReview = values => {
+    if (selfReviewInfo) {
+      dispatch(updateSelfReview(selfReviewInfo._id, values))
+    } else {
+      dispatch(createSelfReview(values))
+    }
+  }
+  return (
```
Comment (shivamsharma30) [inline client/src/components/SelfReviewForm/SelfReviewForm.js:131]: Please break this code into sub components.

File: client/src/sagas/watchers/selfReview.js [hunk 6]
```diff
@@ -32,27 +44,97 @@ function* workerLoadAllUserSelfReviewSaga({ payload }) {
...
     const selfReviews = yield call(updateSelfReview, id, body)
-    if (selfReviews.data.status == 'error') {
+    if (selfReviews.data.status === 'error') {
       yield put(setUpdateReviewError(selfReviews.data.message))
     }
-    if (selfReviews.data.status == 'success') {
+    if (selfReviews.data.status === 'success') {
       yield put(setUpdateReviewStatus(selfReviews.data.message))
     }
+    const Reviews = yield call(loadAllSelfReviews)
```
Comment (shivamsharma30) [inline client/src/sagas/watchers/selfReview.js]: Please use Camel case for variable name.

File: client/src/sagas/watchers/selfReview.js [hunk 7]
```diff
@@ -32,27 +44,97 @@ function* workerLoadAllUserSelfReviewSaga({ payload }) {
...
+  const { body } = payload
+  try {
+    const selfReviews = yield call(createSelfReview, body)
+    if (selfReviews.data.status === 'error') {
+      yield put(setSelfReviewError(selfReviews.data.message))
+    }
+    if (selfReviews.data.status === 'success') {
+      yield put(setSelfReviewSuccess(selfReviews.data.message))
+    }
+    const Reviews = yield call(loadAllSelfReviews)
```
Comment (shivamsharma30) [inline client/src/sagas/watchers/selfReview.js]: Please use Camel case for variable name.

File: client/src/sagas/watchers/selfReview.js [hunk 8]
```diff
@@ -32,27 +44,97 @@ function* workerLoadAllUserSelfReviewSaga({ payload }) {
...
+  const { id } = payload
+  try {
+    const selfReviews = yield call(deleteSelfReview, id)
+    if (selfReviews.data.status === 'error') {
+      yield put(selfReviewDeleteFailue(selfReviews.data.message))
+    }
+    if (selfReviews.data.status === 'success') {
+      yield put(selfReviewDeleteSuccess(selfReviews.data.message))
+    }
+    const Reviews = yield call(loadAllSelfReviews)
```
Comment (shivamsharma30) [inline client/src/sagas/watchers/selfReview.js]: Please use Camel case for variable name.

File: client/src/views/Dashboard/Dashboard.js [hunk 9]
```diff
@@ -61,26 +57,21 @@ const Dashboard = props => {
...
         cls.status !== 'Done' &&
         cls.status !== 'Inactive'
     )
     filteredEmployee.map((review, key) => {
       peerReviewsArray.push([
         review.employee_under_review.firstname +
-          ' ' +
-          review.employee_under_review.lastname,
+        ' ' +
+        review.employee_under_review.lastname,
```
Comment (shivamsharma30) [inline client/src/views/Dashboard/Dashboard.js:74]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/views/PeerReview/PeerReview.js [hunk 10]
```diff
@@ -111,11 +111,11 @@ const PeerReview = props => {
   if (peerReviewData) {
     filteredEmployee = peerReviewData.data.data.filter(
       cls =>
-        cls.employee_under_review.firstname +
+        (cls.employee_under_review.firstname +
           ' ' +
-          cls.employee_under_review.lastname
-            .toLowerCase()
-            .includes(selectedEmployee.toLowerCase().trim()) &&
+          cls.employee_under_review.lastname)
```
Comment (shivamsharma30) [inline client/src/views/PeerReview/PeerReview.js:116]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/views/PeerReview/PeerReview.js [hunk 11]
```diff
@@ -129,8 +129,8 @@ const PeerReview = props => {
             review.employee_under_review.lastname}
         </span>,
         review.employee_reviewing.firstname +
-          ' ' +
-          review.employee_reviewing.lastname,
+        ' ' +
+        review.employee_reviewing.lastname,
```
Comment (shivamsharma30) [inline client/src/views/PeerReview/PeerReview.js:133]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/views/SelfReview/SelfReview.js [hunk 12]
```diff
@@ -1,6 +1,263 @@
...
+                >
+                  <MenuItem className={classes.hoverEffect} value="">
+                    <em>None</em>
+                  </MenuItem>
+                  {employeeData
+                    ? employeeData.map((prop, key) => {
+                      return prop.status !== 'Inactive' ? (
+                        <MenuItem
+                          className={classes.hoverEffect}
+                          value={prop.firstname + ' ' + prop.lastname}
```
Comment (shivamsharma30) [inline client/src/views/SelfReview/SelfReview.js:192]: I think we need to use template literal here. Is it possible to use template literal?

File: client/src/views/SelfReview/SelfReview.js [hunk 13]
```diff
@@ -1,6 +1,263 @@
...
+                          key={key}
+                        >
+                          {prop.firstname} {prop.lastname}
```
Comment (shivamsharma30) [inline client/src/views/SelfReview/SelfReview.js:195]: please Destructuring props then use.

File: client/src/components/Employee/EmployeeListing.js (hunk 2 above)
Comment (RakshandaWankhede) [reply client/src/components/Employee/EmployeeListing.js:176]: DONE

File: client/src/components/Employee/EmployeeListing.js (hunk 3 above)
Comment (RakshandaWankhede) [reply client/src/components/Employee/EmployeeListing.js:253]: DONE

File: client/src/components/Employee/EmployeeListing.js (hunk 4 above)
Comment (RakshandaWankhede) [reply client/src/components/Employee/EmployeeListing.js:218]: DONE

File: client/src/views/Dashboard/Dashboard.js (hunk 9 above)
Comment (RakshandaWankhede) [reply client/src/views/Dashboard/Dashboard.js:74]: did in PR 31

File: client/src/views/PeerReview/PeerReview.js (hunk 10 above)
Comment (RakshandaWankhede) [reply client/src/views/PeerReview/PeerReview.js:116]: Did in PR #31

File: client/src/views/PeerReview/PeerReview.js (hunk 11 above)
Comment (RakshandaWankhede) [reply client/src/views/PeerReview/PeerReview.js:133]: did in PR #31

File: client/src/views/SelfReview/SelfReview.js (hunk 12 above)
Comment (RakshandaWankhede) [reply client/src/views/SelfReview/SelfReview.js:192]: did in PR#31

File: client/src/views/SelfReview/SelfReview.js (hunk 13 above)
Comment (RakshandaWankhede) [reply client/src/views/SelfReview/SelfReview.js:195]: DONE

File: client/src/sagas/watchers/selfReview.js (hunk 6 above)
Comment (RakshandaWankhede) [reply client/src/sagas/watchers/selfReview.js]: done

File: client/src/sagas/watchers/selfReview.js (hunk 7 above)
Comment (RakshandaWankhede) [reply client/src/sagas/watchers/selfReview.js]: done

File: client/src/sagas/watchers/selfReview.js (hunk 8 above)
Comment (RakshandaWankhede) [reply client/src/sagas/watchers/selfReview.js]: done

File: client/src/components/Employee/EmployeeListing.js (hunk 1 above)
Comment (RakshandaWankhede) [reply client/src/components/Employee/EmployeeListing.js:172]: Tried , but not able to optimise

**Comment by RakshandaWankhede** [comment]
same changes are available in PR #31.. closing this PR

**Pull request closed without merging**